src/router/index.js

```diff
@@ -69,25 +70,35 @@ Vue.use(Meta)
 router.beforeResolve((to, from, next) => {
   NProgress.start()
   if (to.name) {
+    let title
+    let workflowName
     if (to.meta.toolbar) {
       // When a workflow is being displayed, we set the title to a
       // different value.
-      store.commit('app/setTitle', to.params.workflowName)
-      store.commit('workflows/SET_WORKFLOW_NAME', { workflowName: to.params.workflowName })
+      title = to.params.workflowName
+      workflowName = to.params.workflowName
     } else {
-      store.commit('app/setTitle', to.name)
-      store.commit('workflows/SET_WORKFLOW_NAME', { workflowName: null })
+      title = to.name
+      workflowName = null
     }
+    store.commit('app/setTitle', title)
+    store.commit('workflows/SET_WORKFLOW_NAME', workflowName)
     store.dispatch('setAlert', null).then(() => {})
   }
   next()
 })
 
 router.beforeEach((to, from, next) => {
   if (!store.state.user.user) {
-    router.app.$userService.getUserProfile().then(() => {
-      next()
-    })
+    router.app.$userService.getUserProfile()
+      .then((user) => {
+        store.commit('user/SET_USER', user)
+        next()
+      })
+      .catch(error => {
+        const alert = new Alert(error, null, 'error')
+        return store.dispatch('setAlert', alert)
+      })
```

Comment:
We already had a dependency/relationship between the router here and the store. Returning the user  in the User Service, eliminates the dependency between User Service and the store, making it easier to test.